src/pages/Trade/components/TradePriceTable/index.tsx

```diff
@@ -190,30 +190,52 @@ const TradePriceTable = () => {
 
   const [currentPrice, setCurrentPrice] = useState('--')
 
+  const getPlaceTableTimer = useRef<ReturnType<typeof setInterval> | undefined>()
```

Comment:
Is the `ref` necessary here? A ref dep by `useEffect(..., [ref])` is awalys the same object

Maybe this is more simple?
```tsx

useEffect(() => {
  function getPriceTable() {
  }
  getPriceTable()

  let intervalId = setInterval(getPriceTable, 5000)

  return () => {
    clearInterval(intervalId)
  }
}, [sudt, lockHash])

```

Reply:
Yes, every time the trade pair changes, we need to clear the previous interval.

Reply:
when the sudt has changed, the `useEffect` will be triggered, the`interval` would be clear like this

`componentDidMount -> useEffect -> sudt changed -> exec useEffect exit function(clearInterval) -> useEffect`

We can try the simple [demo](https://codesandbox.io/s/late-water-7f3s8?file=/src/App.js)

Reply:
Good to know.

Reply:
Good to know.